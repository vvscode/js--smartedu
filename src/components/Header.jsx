import React from "react";

const Header = () => (
  <header className="container mt-3">
    <nav className="navbar navbar-expand-lg navbar-light bg-light">
      <a className="navbar-brand" href="/">
        SmartEdu
      </a>
      <button
        className="navbar-toggler"
        type="button"
        data-toggle="collapse"
        data-target="#navbarNav"
        aria-controls="navbarNav"
        aria-expanded="false"
        aria-label="Toggle navigation"
      >
        <span className="navbar-toggler-icon"></span>
      </button>
      <div className="collapse navbar-collapse" id="navbarNav">
        <ul className="navbar-nav">
          <li className="nav-item active">
            <a className="nav-link" href="/">
              Все курсы
            </a>
          </li>
        </ul>
        <ul className="nav navbar-nav ml-auto">
          <li className="nav-item ">
            <a className="nav-link" href="/contact_us">
              Contact us
            </a>
          </li>

          <li className="nav-item ">
            <a className="nav-link" href="/ajax-register/">
              Register
            </a>
          </li>
          <li className="nav-item ">
            <a className="nav-link" href="/accounts/login/">
              Login
            </a>
          </li>
        </ul>
      </div>
    </nav>
  </header>
);

export default Header;
