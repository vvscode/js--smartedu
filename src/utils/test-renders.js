import React from "react";
import { MemoryRouter as Router } from "react-router";
import { render } from "@testing-library/react";
import { createStore } from "../rdx/createStore";
import { Provider } from "react-redux";

export const renderForTest = (children, preloadedState) => {
  const store = createStore(preloadedState);
  const wrapper = render(
    <Provider store={store}>
      <Router>{children}</Router>
    </Provider>
  );
  return { store, wrapper };
};
