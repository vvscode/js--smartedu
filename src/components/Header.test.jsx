import React from "react";

import Header from "./header";
import { renderForTest } from "../utils/test-renders";

describe("Main Page", () => {
  it("it renders unautorized user menu", () => {
    const { wrapper } = renderForTest(<Header />, {
      user: {
        userInfo: null,
      },
    });
    expect(wrapper).toMatchSnapshot();
  });

  it("it renders autorized user menu", () => {
    const { wrapper } = renderForTest(<Header />, {
      user: {
        userInfo: {
          token: "some token",
        },
      },
    });
    expect(wrapper).toMatchSnapshot();
  });
});
