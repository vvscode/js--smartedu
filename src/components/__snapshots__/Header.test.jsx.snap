// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main Page it renders autorized user menu 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="container mt-3"
      >
        <nav
          class="navbar navbar-expand-lg navbar-light bg-light"
        >
          <a
            class="navbar-brand"
            href="/"
          >
            SmartEdu
          </a>
          <button
            aria-controls="navbarNav"
            aria-expanded="false"
            aria-label="Toggle navigation"
            class="navbar-toggler"
            data-target="#navbarNav"
            data-toggle="collapse"
            type="button"
          >
            <span
              class="navbar-toggler-icon"
            />
          </button>
          <div
            class="collapse navbar-collapse"
            id="navbarNav"
          >
            <ul
              class="navbar-nav"
            >
              <li
                class="nav-item active"
              >
                <a
                  class="nav-link"
                  href="/"
                >
                  Все курсы
                </a>
              </li>
            </ul>
            <ul
              class="nav navbar-nav ml-auto"
            >
              <li
                class="nav-item "
              >
                <a
                  class="nav-link"
                  href="/contact_us"
                >
                  Contact us
                </a>
              </li>
              <li
                class="nav-item"
              >
                <a
                  class="nav-link"
                  href="/profile"
                />
              </li>
              <li
                class="nav-item"
              >
                <a
                  class="nav-link"
                  href="/logout"
                >
                  Logout
                </a>
              </li>
            </ul>
          </div>
        </nav>
      </header>
    </div>
  </body>,
  "container": <div>
    <header
      class="container mt-3"
    >
      <nav
        class="navbar navbar-expand-lg navbar-light bg-light"
      >
        <a
          class="navbar-brand"
          href="/"
        >
          SmartEdu
        </a>
        <button
          aria-controls="navbarNav"
          aria-expanded="false"
          aria-label="Toggle navigation"
          class="navbar-toggler"
          data-target="#navbarNav"
          data-toggle="collapse"
          type="button"
        >
          <span
            class="navbar-toggler-icon"
          />
        </button>
        <div
          class="collapse navbar-collapse"
          id="navbarNav"
        >
          <ul
            class="navbar-nav"
          >
            <li
              class="nav-item active"
            >
              <a
                class="nav-link"
                href="/"
              >
                Все курсы
              </a>
            </li>
          </ul>
          <ul
            class="nav navbar-nav ml-auto"
          >
            <li
              class="nav-item "
            >
              <a
                class="nav-link"
                href="/contact_us"
              >
                Contact us
              </a>
            </li>
            <li
              class="nav-item"
            >
              <a
                class="nav-link"
                href="/profile"
              />
            </li>
            <li
              class="nav-item"
            >
              <a
                class="nav-link"
                href="/logout"
              >
                Logout
              </a>
            </li>
          </ul>
        </div>
      </nav>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Main Page it renders unautorized user menu 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="container mt-3"
      >
        <nav
          class="navbar navbar-expand-lg navbar-light bg-light"
        >
          <a
            class="navbar-brand"
            href="/"
          >
            SmartEdu
          </a>
          <button
            aria-controls="navbarNav"
            aria-expanded="false"
            aria-label="Toggle navigation"
            class="navbar-toggler"
            data-target="#navbarNav"
            data-toggle="collapse"
            type="button"
          >
            <span
              class="navbar-toggler-icon"
            />
          </button>
          <div
            class="collapse navbar-collapse"
            id="navbarNav"
          >
            <ul
              class="navbar-nav"
            >
              <li
                class="nav-item active"
              >
                <a
                  class="nav-link"
                  href="/"
                >
                  Все курсы
                </a>
              </li>
            </ul>
            <ul
              class="nav navbar-nav ml-auto"
            >
              <li
                class="nav-item "
              >
                <a
                  class="nav-link"
                  href="/contact_us"
                >
                  Contact us
                </a>
              </li>
              <li
                class="nav-item"
              >
                <a
                  class="nav-link"
                  href="/ajax-register/"
                >
                  Register
                </a>
              </li>
              <li
                class="nav-item"
              >
                <a
                  class="nav-link"
                  href="/accounts/login/"
                >
                  Login
                </a>
              </li>
            </ul>
          </div>
        </nav>
      </header>
    </div>
  </body>,
  "container": <div>
    <header
      class="container mt-3"
    >
      <nav
        class="navbar navbar-expand-lg navbar-light bg-light"
      >
        <a
          class="navbar-brand"
          href="/"
        >
          SmartEdu
        </a>
        <button
          aria-controls="navbarNav"
          aria-expanded="false"
          aria-label="Toggle navigation"
          class="navbar-toggler"
          data-target="#navbarNav"
          data-toggle="collapse"
          type="button"
        >
          <span
            class="navbar-toggler-icon"
          />
        </button>
        <div
          class="collapse navbar-collapse"
          id="navbarNav"
        >
          <ul
            class="navbar-nav"
          >
            <li
              class="nav-item active"
            >
              <a
                class="nav-link"
                href="/"
              >
                Все курсы
              </a>
            </li>
          </ul>
          <ul
            class="nav navbar-nav ml-auto"
          >
            <li
              class="nav-item "
            >
              <a
                class="nav-link"
                href="/contact_us"
              >
                Contact us
              </a>
            </li>
            <li
              class="nav-item"
            >
              <a
                class="nav-link"
                href="/ajax-register/"
              >
                Register
              </a>
            </li>
            <li
              class="nav-item"
            >
              <a
                class="nav-link"
                href="/accounts/login/"
              >
                Login
              </a>
            </li>
          </ul>
        </div>
      </nav>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
